File: ios/Modules/Sources/Push/api/PushSubscriber.swift
import Foundation

/// APNs push subscription. Implementation calls `application.registerForRemoteNotifications()`
/// + posts the device token (plus a generated X25519 pubkey + 16-byte auth
/// secret) to `/api/push/subscribe` so the node can encrypt payloads
/// end-to-end even through Apple's servers.
public protocol PushSubscriber: Sendable {
    func enable() async -> Bool
    func disable() async -> Bool
}

File: android/app/src/main/java/sol/vortexx/android/push/api/PushSubscriber.kt
package sol.vortexx.android.push.api

/**
 * Wave 18 push notifications.
 *
 * Implementation registers with FCM, obtains the token, posts it (plus the
 * VAPID server key the node advertises) to `/api/push/subscribe`. The
 * server encrypts the push payload with the subscription's p256dh+auth
 * pair so even Google cannot read the content.
 */
interface PushSubscriber {
    suspend fun enable(): Boolean
    suspend fun disable(): Boolean
}
